### ISSUE TITLE
Fix cache option hiding CssSyntaxError outputs

### ISSUE THREAD

**Issue by otakustay** [stylelint/stylelint#3254]
When a css file contains syntax errors like:

```css
a {
  {color: red;
}
```

When `cache` option is set to `true`, stylelint will report a `CssSyntaxError` first time it runs, however starting from the second time, when css file is not modified, `CssSyntaxError` disappears from output

**Comment by jeddy3** [comment]
@otakustay Thanks for the report.

@sergesemashko I assume this isn't the intended behaviour?

**Comment by jeddy3** [comment]
> From a brief look, it looks like files are getting removed from cache only on style violations, but not in case of syntax error.

Thanks for confirming the issue. I'll label this up accordingly ("bug" and "help wanted").

@otakustay Please consider contributing a fix if you'd like to see this issue resolved as the stylelint core team is small.

**Comment by sergesemashko** [comment]
@otakustay, let me know if you decide to fix it yourself. Otherwise, I'll take a look at the issue this week when have a moment

**Comment by otakustay** [comment]
@sergesemashko I'm going to dig into this issue tomorrow, if I find myself unable to solve it, will pin you again :)
